test/hw9/test_task2.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+from All_home_works.hw9.task2 import Suppressor
+
+
+def test_class_suppressor__catch_expected_exception():
+    with Suppressor(IndexError):
+        [][2]
+    assert True
+
+
+def test_class_suppressor__catch_not_expected_exception():
+    with pytest.raises(IndexError, match="list index out of range"):  # noqa: PT012
+        with Suppressor(ZeroDivisionError):
+            [][2]
+
+
+def test_generator_suppressor__catch_expected_exception():
+    with Suppressor(IndexError):
+        [][2]
+    assert True
+
+
+def test_generator_suppressor__catch_not_expected_exception():
+    with pytest.raises(IndexError, match="list index out of range"):  # noqa: PT012
```

Comment:
What is PT012?

Reply:
PT012 pytest.raises() block should contain a single simple statement